src/level/robotmain.cpp

```diff
@@ -5482,93 +5483,47 @@ void CRobotMain::SetAutosaveSlots(int slots)
     if (m_autosaveSlots == slots) return;
 
     m_autosaveSlots = slots;
-    AutosaveRotate(false);
+    AutosaveRotate();
 }
 
 int CRobotMain::GetAutosaveSlots()
 {
     return m_autosaveSlots;
 }
 
-int CRobotMain::AutosaveRotate(bool freeOne)
+// Remove oldest saves with autosave prefix
+void CRobotMain::AutosaveRotate()
```

Comment:
Removing the freeOne parameter broke behaviour of autosave rotation when changing amount of autosave slots in settings. Now when you adjust the number of slots, the AutosaveRotate function will remove one save too much (it assumes that a new save is going to be always created immediately after, which is not the case)